stdlib/ballerina-mime/src/main/java/org/ballerinalang/mime/util/EntityBodyHandler.java

```diff
@@ -206,11 +219,23 @@ public static BJSON constructJsonDataSource(BStruct entityStruct) {
      */
     public static BXML constructXmlDataSource(BStruct entityStruct) {
         try {
+            BXML xmlContent;
             Channel byteChannel = getByteChannel(entityStruct);
             if (byteChannel == null) {
                 throw new BallerinaIOException("Empty xml payload");
             }
-            BXML xmlContent = XMLUtils.parse(byteChannel.getInputStream());
+            String contentTypeValue = HeaderUtil.getHeaderValue(entityStruct,
+                    HttpHeaderNames.CONTENT_TYPE.toString());
+            if (contentTypeValue != null && !contentTypeValue.isEmpty()) {
+                String charsetValue = MimeUtil.getContentTypeParamValue(contentTypeValue, CHARSET);
+                if (charsetValue != null && !charsetValue.isEmpty()) {
```

Comment:
I think it would make sense to move this null/empty check logic to the `parse()` method.. Because we have to do this check every time we use this method..

Same applies for JSON and string parse methods

Reply:
I don't think it is wise to move the null and empty check to parse method because the action (using default charset in case of null or empty) will not what the parse method user anticipate always. We should keep the implementation as it is.